test/Dialect/FIRRTL/emit-metadata.mlir

```diff
@@ -0,0 +1,98 @@
+
+//===----------------------------------------------------------------------===//
+// SitestBlackbox
+//===----------------------------------------------------------------------===//
+
+firrtl.circuit "Test" attributes { annotations = [{
+    class = "sifive.enterprise.firrtl.SitestBlackBoxAnnotation",
+    filename = "dut_blackboxes.json"
+  }]} {
+  firrtl.module @Test() attributes {annotations = [
+      {class = "sifive.enterprise.firrtl.MarkDUTAnnotation"}]} {
+  }
+}
+// CHECK:      firrtl.circuit "Test"   {
+// CHECK-NEXT:   firrtl.module @Test() attributes {annotations = [{class = "sifive.enterprise.firrtl.MarkDUTAnnotation"}]} {
+// CHECK-NEXT:   }
+// CHECK-NEXT:   sv.verbatim "[]" {output_file = #hw.output_file<"dut_blackboxes.json", excludeFromFileList>, symbols = []}
+// CHECK-NEXT: }
+
+firrtl.circuit "Test" attributes { annotations = [{
+    class = "sifive.enterprise.firrtl.SitestTestHarnessBlackBoxAnnotation",
+    filename = "test_blackboxes.json"
+  }]} {
+  firrtl.module @Test() attributes {annotations = [
+      {class = "sifive.enterprise.firrtl.MarkDUTAnnotation"}]} {
+  }
+}
+// CHECK-NEXT: firrtl.circuit "Test"   {
+// CHECK-NEXT:   firrtl.module @Test() attributes {annotations = [{class = "sifive.enterprise.firrtl.MarkDUTAnnotation"}]} {
+// CHECK-NEXT:   }
+// CHECK-NEXT:   sv.verbatim "[]" {output_file = #hw.output_file<"test_blackboxes.json", excludeFromFileList>, symbols = []}
+// CHECK-NEXT: }
+
+// CHECK: firrtl.circuit "Test"   {
+firrtl.circuit "Test" attributes { annotations = [{
+    class = "sifive.enterprise.firrtl.SitestBlackBoxAnnotation",
+    filename = "dut_blackboxes.json"
+  }, {
+    class = "sifive.enterprise.firrtl.SitestTestHarnessBlackBoxAnnotation",
+    filename = "test_blackboxes.json"
+  }]} {
+
+  firrtl.module @Test() attributes {annotations = [
+      {class = "sifive.enterprise.firrtl.MarkDUTAnnotation"}]} {
+    firrtl.instance @DUTBlackbox { name = "test" }
+  }
+
+  // These should all be ignored.
+  firrtl.extmodule @ignored0() attributes {annotations = [{class = "firrtl.transforms.BlackBoxInlineAnno"}], defname = "ignored0"}
+  firrtl.extmodule @ignored1() attributes {annotations = [{class = "firrtl.transforms.BlackBoxResourceAnno"}], defname = "ignored1"}
+  firrtl.extmodule @ignored2() attributes {annotations = [{class = "sifive.enterprise.firrtl.ScalaClassAnnotation", className = "freechips.rocketchip.util.BlackBoxedROM"}], defname = "ignored2"}
+
+  // Gracefully handle missing defnames.
+  firrtl.extmodule @NoDefName()
+
+  firrtl.extmodule @TestBlackbox() attributes {defname = "TestBlackbox"}
+  // CHECK: sv.verbatim "[\22TestBlackbox\22]" {output_file = #hw.output_file<"test_blackboxes.json", excludeFromFileList>, symbols = []}
+
+  firrtl.extmodule @DUTBlackbox() attributes {defname = "DUTBlackbox"}
+  // CHECK: sv.verbatim "[\22DUTBlackbox\22]" {output_file = #hw.output_file<"dut_blackboxes.json", excludeFromFileList>, symbols = []}
+}
```

Comment:
I would include a `CHECK-NOT: sv.verbatim` to check the ignored stuff above.

Reply:
They all get printed in a JSON array in the same `sv.verbatim`, e.g. when something goes wrong it looks like this `"[\22ignored2\22, \22DUTBlackbox\22]"`, so I think this is good to catch that the ignored stuff is ignored.